rs/replicated_state/src/canister_state/queues/message_pool.rs

```diff
@@ -117,6 +117,18 @@ impl Id {
             Class::BestEffort
         }
     }
+
+    /// Tests whether this `Id` represents an inbound best-effort response.
+    fn is_inbound_best_effort_response(&self) -> bool {
```

Comment:
Nice! I like that this is encapsulated now as it makes the places in the code where this is checked much easier to read.

Do we need (or maybe even already have) some tests around these functions? It would appear that they are easy to break in subtle ways.